cmake/CMakeLists.txt

```diff
@@ -144,16 +144,27 @@ if (MSVC)
   endif()
   #Always enable exception handling, even for Windows ARM
   SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
+  SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /Zi")
+  SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zi")
   #Disable 4100 globally. Too many this kind errors in protobuf
   SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4100")
   if (NOT onnxruntime_USE_CUDA)
     SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /GL")
     SET (CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} /GL")
   endif()
-  if (onnxruntime_ENABLE_QSPECTRE)
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /Qspectre /guard:cf /Zi")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Qspectre /guard:cf /Zi")
-    SET(CMAKE_EXE_LINKER_FLAGS  "/guard:cf /DYNAMICBASE")
+  check_cxx_compiler_flag(-Qspectre HAS_QSPECTRE)
```

Comment:
Does this also check that the toolset is installed? I think compiler switch should be present -- not sure it this checks if the spectre libs are installed. Worth double checking.

Reply:
It checked it.  I verified.